### ISSUE TITLE
Show estimated CO2 footprint alongside tokens and cost

### ISSUE THREAD

**Issue by ingo-eichhorst** [ingo-eichhorst/Irrlicht#829]
## Summary

Show an estimated CO2 footprint per session, as a third selectable/cyclable metric alongside the existing token count and dollar cost (mirrors the existing tokens ↔ cost display pattern already used across surfaces). This is necessarily an **estimate**, not a measurement — no provider exposes real per-request energy telemetry — so the feature must ship with a visible confidence tier and a methodology link, not a bare number presented as fact.

## Why this is feasible despite no provider API for it

No major LLM provider publishes a live per-request CO2 API, but there's enough public data to build a defensible tiered estimate:

- **Google** (Aug 2025, [Cloud blog](https://cloud.google.com/blog/products/infrastructure/measuring-the-environmental-impact-of-ai-inference), [full report](https://services.google.com/fh/files/misc/measuring_the_environmental_impact_of_delivering_ai_at_google_scale.pdf)): median Gemini text prompt = 0.24 Wh / **0.03 gCO2e** (comprehensive accounting incl. idle capacity + cooling).
- **Mistral** + Carbone4/ADEME (Jul 2025, peer-reviewed LCA, [Carbone4](https://www.carbone4.com/en/ia-generative-mission-mistral-ai), [Mistral](https://mistral.ai/news/our-contribution-to-a-global-environmental-standard-for-ai/)): Mistral Large 2 marginal inference = **1.14 gCO2e per 400-token prompt** (≈2.85 mgCO2e/token, full lifecycle incl. amortized training).
- **OpenAI** (Jun 2025, Altman blog, [coverage](https://www.datacenterdynamics.com/en/news/sam-altman-chatgpt-queries-consume-034-watt-hours-of-electricity-and-0000085-gallons-of-water/)): ~0.34 Wh/query — energy only, no gCO2e, no disclosed methodology.
- **Anthropic**: no public per-query disclosure found.
- **[EcoLogits](https://github.com/genai-impact/ecologits)** (Data For Good + Boavizta, MIT-licensed, Python): open formula + coefficient table estimating GPU energy from token counts and (where known) active parameter count, converted to CO2e via per-country grid factors. Solid for open-weight models; OpenAI/Anthropic entries are flagged as rough estimates since real param counts aren't public — useful as a reference implementation to port the *shape* of, not as an authoritative table for the closed models irrlicht mostly tracks.
- **[Epoch AI](https://epoch.ai/gradient-updates/how-much-energy-does-chatgpt-use)** (Feb 2025) revised frontier-model energy down to ~0.3 Wh/query median (IQR 0.16–0.60) — supersedes older 3 Wh/query estimates (de Vries 2023) that are now considered 4–20x too high. Cross-model spread confirmed by Jegham et al., ["How Hungry is AI?"](https://arxiv.org/abs/2505.09598) (0.4–29+ Wh/query across 30 models).
- **Grid carbon intensity** (needed to turn Wh into gCO2e): global average ~458–473 gCO2/kWh ([Ember 2025](https://ember-energy.org/latest-insights/global-electricity-review-2025/)), US ~373–386 ([EPA eGrid](https://www.epa.gov/energy/greenhouse-gas-equivalencies-calculator-calculations-and-references)), EU ~180–240 ([EEA](https://www.eea.europa.eu/en/analysis/indicators/greenhouse-gas-emission-intensity-of-1)) — region alone swings the estimate ~2x, so use a global-average constant by default rather than guessing a datacenter region. Live data is available from [Electricity Maps](https://app.electricitymaps.com/docs) (free tier) if we ever want to get fancier.
- **Prior art for the "it's an estimate" disclosure pattern**: [EcoLogits' browser extension](https://github.com/genai-impact/ecologits), carbon-llm.com, [antarctica.io's One Token Model](https://antarctica.io/research/one-token-model), aiimpactcalculator.com — all label figures as estimates with a methodology link. No mainstream product (ChatGPT/Gemini/Claude UI) shows this today, so irrlicht doing it with honest confidence-tiering is genuinely differentiated, not just copying an existing feature.

## Proposed formula

```
CO2e_grams = (input_tokens·e_in + output_tokens·e_out + cache_tokens·e_cache) × PUE × grid_gCO2_per_kWh / 1000
```

Where `e_*` is Wh/token for a given model, looked up through **three confidence tiers** (ship the tier alongside the number, don't collapse it into one undifferentiated figure):

1. **Provider-disclosed** — normalize Google's/Mistral's published per-prompt figures to Wh/token for the specific models they cover.
2. **EcoLogits-style estimate** — for models with reasonably-known/leaked active-parameter counts.
3. **Generic fallback** — ≈0.0003 Wh/token (Epoch AI's ~0.3 Wh/query ÷ ~1k-token assumed query length) × global grid average, for anything without tiers 1–2. This is the tier most irrlicht-tracked sessions (Claude, GPT-5, Gemini via CLI agents) will fall into today — say so plainly in the UI rather than implying precision the source data doesn't have.

Default `PUE` 1.1–1.2 (EcoLogits' disclosed provider range) and default grid intensity = global average (~460 gCO2/kWh) unless/until we have per-provider region data.

## Where this plugs into the existing codebase

Cost is computed and surfaced through a well-established path — CO2 should follow the same shape:

- **Coefficient table**: `core/pkg/capacity/capacity.go:10-27` (`ModelPricing` struct) is the existing per-model-coefficient pattern (USD per million tokens, keyed by model name in `capacityConfig.Models`). Pricing itself is fetched remotely from LiteLLM's `model_prices_and_context_window.json` (`core/pkg/capacity/remote.go:21`, cached 24h) — LiteLLM has **no CO2/energy field**, so the CO2 coefficient table needs its own data source; simplest path is a small hardcoded/embedded Go table (tier 1/2 entries for the handful of models we have real numbers for, tier-3 fallback for everything else), not another remote fetch.
- **Computation**: mirror `(*CapacityManager).EstimateCostUSD` (`core/pkg/capacity/capacity.go:157-169`) with an `EstimateCO2Grams(modelName, inputTokens, outputTokens, cacheReadTokens, cacheCreationTokens int64) (grams float64, tier ConfidenceTier)`. Called from the same place cost is: `computeCumulativeTokens()` in `core/pkg/tailer/tailer_metrics.go:241-299`.
- **Domain type**: add `EstimatedCO2Grams` (+ tier) next to `EstimatedCostUSD` in `SessionMetrics` (`core/domain/session/metrics.go:105`), carried forward in `MergeMetrics` (`:410-422`) same as cost.
- **Surfaces to update** (all currently show tokens/cost, per this pattern):
  - CLI: `core/cmd/irrlicht-ls/render.go:165` (`formatCostUSD`) — add a `formatCO2` alongside it.
  - Web dashboard: `platforms/web/irrlicht.js:656` (`formatCost`), `:811` (tokens) — there's already a `cycleCostTimeframe()` pattern (`:540-545`) for cycling a displayed dimension on click; add CO2 as a third value in that style of selector rather than a fourth always-on column.
  - macOS app: `platforms/macos/Irrlicht/Models/SessionState.swift:350` (tokens), `:356` (cost), JSON keys `:371,377`.
  - Daemon JSON API: `core/cmd/irrlichd/handlers.go:823,1059,1085,1097`.
  - Persisted cost history: `core/adapters/outbound/filesystem/cost_tracker.go:41-53` (`snapshotRow`) — add a CO2 field so historical charts can show it too.

## Non-goals / caveats

- Not a measurement — every number is a modeled estimate with cited assumptions. UI must show the confidence tier and link the methodology (short doc or `--help` text), not present a bare gram figure.
- No live grid-intensity API integration in v1 — global-average constant is enough to start; region-aware intensity (Electricity Maps) is a reasonable follow-up, not a blocker.
- Anthropic/OpenAI-specific tier-1 numbers don't exist publicly — don't block on getting them; ship with tier-3 fallback for those and revisit if either publishes real figures.

## Suggested scope for a first PR

1. Coefficient table + `EstimateCO2Grams` in `core/pkg/capacity` + unit tests (formula correctness, tier selection, fallback).
2. Wire into `SessionMetrics` + `tailer_metrics.go` cumulative computation.
3. Surface in one place first (web dashboard, since it already has the `cycleCostTimeframe` selector pattern to extend) to validate the UX, then follow up with CLI/macOS/daemon-API parity.
